cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/services/PropertiesResolver.java

```diff
@@ -35,11 +35,13 @@
 
   private final PreferencesStore prefStore;
   private final SchedulerQueueResolver queueResolver;
+  private final ImpersonationUserResolver impersonationUserResolver;
 
   @Inject
   PropertiesResolver(PreferencesStore prefStore, NamespaceStore store, CConfiguration cConf) {
     this.prefStore = prefStore;
     this.queueResolver = new SchedulerQueueResolver(cConf, store);
+    this.impersonationUserResolver = new ImpersonationUserResolver(cConf, store);
```

Comment:
Why new it? The `ImpersonationUserResolver` should have a `@Inject` constructor. BTW, will it be possible that there can be more one implementation of the `ImpersonationUserResolver`?